File: code/Java/interpreter/expr/ReadIntExpr.java
package interpreter.expr;

import java.util.Scanner;

public class ReadIntExpr extends IntExpr {

    private static Scanner in;
    
    static {
        in = new Scanner(System.in);
    }

    public ReadIntExpr(int line) {
        super(line);
    }

    public int expr() {
        try {
            String str = in.nextLine();
            int n = Integer.parseInt(str);
            return n;
        } catch (Exception e) {
            return 0;
        }
    }

}

File: code/C++/interpreter/expr/ReadIntExpr.cpp
#include <iostream>
#include "ReadIntExpr.h"

ReadIntExpr::ReadIntExpr(int line)
	: IntExpr(line) {
}

ReadIntExpr::~ReadIntExpr() {
}

int ReadIntExpr::expr() {
	int value;
	std::cin >> value;
	return value;
}
